driver-core/src/test/java/com/datastax/driver/core/DurationCodecTest.java

```diff
@@ -0,0 +1,92 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.core;
+
+import org.testng.annotations.Test;
+
+import java.nio.ByteBuffer;
+
+import static com.datastax.driver.core.Duration.*;
+import static com.datastax.driver.core.ProtocolVersion.V4;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class DurationCodecTest {
+
+    @Test(groups = "unit")
+    public void testFromStringWithStandardPattern() {
+        assertCodec("1y2mo", Duration.newInstance(14, 0, 0));
+        assertCodec("-1y2mo", Duration.newInstance(-14, 0, 0));
```

Comment:
It is actually illegal to mix positive and negative attributes for a C* Duration. C* doesn't complain about it when you use bound parameters, but you can't feed literals like this in cql. Ben Lerer suggested that drivers validate the data because it is expensive for C* to validate on its end (though he said he'll mull over an efficient way to reject server-side).